test/index.js

```diff
@@ -1,6 +1,46 @@
-import {expect} from 'chai';
-import works from '../src';
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import stipulate from '../src/stipulate';
+
+chai.use(chaiAsPromised);
+
+// error handling tested elsewhere; bypass it
+const okBypass = () => true;
+// mock response that can respond to different data
+// extraction methods.
+const responseMock = (url, options) => {
+  const jsonData = Promise.resolve([url, options]);
```

Comment:
This is slightly confusing. Can you make this some dummy value that can be asserted in the test?

My confusion:
```
const results = stipulate(url, opts);
return expect(results).to.eventually.deep.equal([url, opts]);
```

Makes me think that stipulate is supposed to return what you pass in.

Instead, if it was:
```
const resource = stipulate(url, opts);
return expect(resource).to.eventually.deep.equal({some:'dummy data'});
```

then this would be perfectly clear.